src/commands/timelinecommands.cpp

```diff
@@ -1307,22 +1310,38 @@ DetachAudioCommand::DetachAudioCommand(MultitrackModel& model, int trackIndex, i
 void DetachAudioCommand::redo()
 {
     LOG_DEBUG() << "trackIndex" << m_trackIndex << "clipIndex" << m_clipIndex << "position" << m_position;
-    Mlt::Producer clip(MLT.profile(), "xml-string", m_xml.toUtf8().constData());
-    if (clip.is_valid()) {
-        // Save the original clip's audio_index.
-        m_audioIndex = QString::fromLatin1(clip.get("audio_index"));
-
-        // Disable audio on the original clip.
-        int i = m_model.trackList().at(m_trackIndex).mlt_index;
-        Mlt::Producer track(m_model.tractor()->track(i));
-        if (track.is_valid()) {
-            Mlt::Playlist playlist(track);
-            Mlt::Producer originalClip(playlist.get_clip(m_clipIndex));
-            if (originalClip.is_valid()) {
-                originalClip.parent().set("audio_index", -1);
-                QModelIndex modelIndex = m_model.makeIndex(m_trackIndex, m_clipIndex);
-                emit m_model.dataChanged(modelIndex, modelIndex, QVector<int>() << MultitrackModel::AudioIndexRole);
+    Mlt::Producer audioClip(MLT.profile(), "xml-string", m_xml.toUtf8().constData());
+    Mlt::Producer videoClip(MLT.profile(), "xml-string", m_xml.toUtf8().constData());
+    if (audioClip.is_valid() && videoClip.is_valid()) {
+
+        // Disable audio on the video clip.
+        videoClip.set("audio_index", -1);
+        // Remove audio filters from the video clip.
+        for (int i = 0; i < videoClip.filter_count(); i++) {
+            Mlt::Filter* filter = videoClip.filter(i);
+            if (filter && filter->is_valid() && !filter->get_int("_loader")) {
+                QmlMetadata* newMeta = MAIN.filterController()->metadataForService(filter);
+                if (newMeta && newMeta->isAudio()) {
+                    videoClip.detach(*filter);
+                    i--;
+                }
+            }
+            delete filter;
+        }
+
+        // Disable video on the audio clip.
+        audioClip.set("video_index", -1);
```

Comment:
Video is already hidden on audio tracks. It is not advisable to change the video_index because if it was a video clip and the user drags it back from audio to video track, they need to turn the video back on.

Reply:
> if it was a video clip and the user drags it back from audio to video track, they need to turn the video back on.

Just to make sure we are talking about the same thing...

The "Detach Audio" feature does not convert a video+audio clip to an audio only clip. It duplicates the clip and leaves the video clip on the original video track and the audio clip on another track. So they can't move the audio clip "back" since the original video clip is still there.

I'm happy to remove line 1333, just want to make sure we both have the same understanding of the change.